pkg/clustermesh/types/option.go

```diff
@@ -66,6 +68,20 @@ func (c ClusterInfo) ValidateStrict(log logrus.FieldLogger) error {
 	return c.validateName(log)
 }
 
+// ValidateBuggyClusterID returns an error if a buggy cluster ID (i.e., with the
+// 7th bit set) is used in combination with ENI IPAM mode or AWS CNI chaining.
+func (c ClusterInfo) ValidateBuggyClusterID(ipamMode, chainingMode string) error {
+	if (c.ID&0x80) != 0 && (ipamMode == ipamOption.IPAMENI || ipamMode == ipamOption.IPAMAlibabaCloud || chainingMode == "aws-cni") {
```

Comment:
I think that the reason that ENI mode is affected is not necessarily to do with the mode, but more that the current default instructions for installing Cilium in EKS is to first deploy EKS (which deploys aws-vpc-cni, creating the bug), then disable aws-vpc-cni.

I don't know if AlibabaCloud has the same behaviour where the default installation of the managed cloud environment will create a conflicting iptables rule as described in #21330. Maybe @jaffcheng or @l1b0k  could speak to that.

I anticipate that when we migrate the ENI instructions over to BYOCNI we should be able to lift this restriction specifically on Cilium ENI (when deployed directly without aws-vpc-cni). But for now it's probably safer to fail out, and we can deal with making this restriction looser over time.

Reply:
> I think that the reason that ENI mode is affected is not necessarily to do with the mode, but more that the current default instructions for installing Cilium in EKS is to first deploy EKS (which deploys aws-vpc-cni, creating the bug), then disable aws-vpc-cni.

Based on:

> Cilium is running with ENI mode without AWS VPC CNI from the beginning (e.g. self-hosted k8s cluster on EC2 hosts)

the bug appears to affect Cilium in ENI mode without AWS VPC CNI as well. That's because of the iptables rules that get added [here](https://github.com/cilium/cilium/blob/3873b81b2ef5c31ab37471ac41f473cb7fb68537/pkg/datapath/iptables/iptables.go#L1764-L1797) when Cilium is configured in ENI mode. I've added AlibabaCloud as well because the rules [are also added in that case](https://github.com/cilium/cilium/blob/3873b81b2ef5c31ab37471ac41f473cb7fb68537/pkg/datapath/iptables/iptables.go#L1565-L1570), although it is not clear to be whether they are actually necessary or not.

And I've reproduced the issue with Cilium running in ENI mode, and no AWS-CNI iptables rules at all.

Reply:
Ah okay, makes sense yep. I misread the corresponding issue. :+1: